core/src/main/java/io/grpc/internal/DnsNameResolver.java

```diff
@@ -303,26 +303,29 @@ public void run() {
         return;
       }
 
-      Attributes.Builder attrs = Attributes.newBuilder();
+      ResolutionResult.Builder resultBuilder = ResolutionResult.newBuilder().setAddresses(servers);
       if (!resolutionResults.txtRecords.isEmpty()) {
         ConfigOrError serviceConfig =
             parseServiceConfig(resolutionResults.txtRecords, random, getLocalHostname());
         if (serviceConfig != null) {
           if (serviceConfig.getError() != null) {
-            savedListener.onError(serviceConfig.getError());
+            savedListener.handleError(resultBuilder.setServiceConfig(serviceConfig).build());
```

Comment:
This is the "invalid config" case. Default config may be used in such case. You should still send the `ResolutionResult` to `savedListener.onResult()`, and let the error be handled by `ManagedChannelImpl` who has access to the default config. `savedListener.onError()` is intended for name resolution error, which is not the case here.

Reply:
thanks! you are right, onError should be the name resolution error. 
onResult calls onError if the invalid serviceConfig ends up cause the channel transient failure. PTAL